### PR TITLE
example using multiple indicators inside of an algorithm. upgraded indicators to support the entire dictionary of cached datasets to enable calculations across daily + minute + options + news to build custom indicators from any of the data found from the redis data pipeline

### PR DESCRIPTION
**Pull request by jay-johnson**

### CODE REVIEW

File: analysis_engine/indicators/base_indicator.py [hunk 1]
```diff
@@ -229,6 +217,70 @@ def get_name(
...
+            supported ``pd.DataFrame`` that is in the
+            cached ``dataset['data']`` dictionary
+            of dataframes
+        """
+        return dataset['data'].get(
+            dataset_name,
+            pd.DataFrame(ae_consts.EMPTY_DF_LIST))
+    # end of get_dataset_by_name
+
+    def get_subcribed_dataset(
```
Comment (42B) [inline analysis_engine/indicators/base_indicator.py]: "subcribed" -> "subscribed"?

File: analysis_engine/indicators/base_indicator.py [hunk 2]
```diff
@@ -229,6 +217,70 @@ def get_name(
...
+            self,
+            dataset,
+            dataset_name=None):
+        """get_subcribed_dataset
```
Comment (42B) [inline analysis_engine/indicators/base_indicator.py]: "subcribed" -> "subscribed"?

File: analysis_engine/mocks/example_indicator_williamsr.py [hunk 3]
```diff
@@ -90,27 +89,31 @@ def process(
         log.info(
             '{} - start'.format(
                 label))
+
+        daily_df = self.get_subcribed_dataset(
```
Comment (42B) [inline analysis_engine/mocks/example_indicator_williamsr.py]: "subcribed" -> "subscribed"?

File: analysis_engine/mocks/example_indicator_williamsr_open.py [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+        label = (
+            '{} process ticker={}'.format(
+                self.name,
+                ticker))
+
+        log.info(
+            '{} - start'.format(
+                label))
+
+        daily_df = self.get_subcribed_dataset(
```
Comment (42B) [inline analysis_engine/mocks/example_indicator_williamsr_open.py]: "subcribed" -> "subscribed"?

File: analysis_engine/indicators/base_indicator.py (hunk 1 above)
Comment (jay-johnson) [reply analysis_engine/indicators/base_indicator.py]: nice catch! thanks for that!

me fail english

Comment (42B) [reply analysis_engine/indicators/base_indicator.py]: @jay-johnson 

Haha no problem!

**Pull request merged**